packages/@aws-cdk/cdk/lib/cloudformation/pseudo.ts

```diff
@@ -40,10 +83,6 @@ export class Aws {
   public get stackName(): string {
     return new AwsStackName(this.scope).toString();
```

Comment:
How does it make sense to export { AWS::StackName }? Or is it just to discover that things come from a different stack?

Something is oddd

Reply:
It *could* make sense:

```ts
Properties:
  Thing: 
    Type: AWS::Thing::Thing
Outputs:
  ThingStack:
    Value: { Ref: "AWS::StackName" }
    Export: 
      Name: TheStackWithTheThing
```

Creates an export with a well-known name pointing to the name under which this stack was ultimately deployed (does not necessarily need to be the name known to the CDK at deployment time).

Or do you mean, why can `stackName` be scoped? Because then the  answer is, yes, to detect that it came from a different `Stack` object.

```ts
const stack1 = new StackWithThing(...);

const stack2 = new Stack(...);
new ThingConsumer(..., {
   thingStackName: stack1.stackName // would not be the right value if nonscoped
});

```

Reply:
got it, thanks